exercises/prime-factors/canonical-data.json

```diff
@@ -0,0 +1,42 @@
+{
+  "for": {
+    "description": "returns prime factors for the given input number",
+    "cases": [
+      {
+        "description" : "no factors",
+        "input" : 1,
+        "expected" : []
+      },
+      {
+        "description" : "prime number",
+        "input" : 2,
+        "expected" : [2]
+      },
+      {
+        "description" : "square of a prime",
+        "input" : 9,
+        "expected" : [3, 3]
+      },
+      {
+        "description" : "cube of a prime",
+        "input" : 8,
+        "expected" : [2, 2, 2]
+      },
+      {
+        "description" : "product of primes and non-primes",
+        "input" : 12,
+        "expected" : [2, 2, 3]
+      },
+      {
+        "description" : "product of primes",
+        "input" : 901255,
+        "expected" : [5, 17, 23, 461]
+      },
+      {
+        "description" : "factors include a large prime",
+        "input" : 93819012551,
+        "expected" : [11, 9539, 894119]
```

Comment:
caveat: I've not tried this problem myself recently. 

Are these numbers (93819012551, 901255) within the bounds of reasonableness (<1sec) for today's computers without having to rely on Project Euler style optimisation trickery?